openlibrary/macros/UserMetadata.html

```diff
@@ -1,25 +1,37 @@
-$def with (work, edition, link_text, id, classes='')
+$def with (link_text, id, classes='', work=None, edition=None, notes=None, work_key=None, reload_id=None)
```

Comment:
Maybe work and edition can be combined as `page` (or `doc`).  Then we can determine if the object is an edition or a work.